src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Debug/ILaunchSettingsProvider.cs

```diff
@@ -13,6 +13,11 @@ namespace Microsoft.VisualStudio.ProjectSystem.Debug
     [ProjectSystemContract(ProjectSystemContractScope.UnconfiguredProject, ProjectSystemContractProvider.Private, Cardinality = ImportCardinality.ExactlyOne)]
     public interface ILaunchSettingsProvider
     {
+        /// <remarks>
+        /// If the <see cref="ILaunchSettings"/> provided by this block are going to feed
+        /// into another data flow block, strongly consider using <see cref="IVersionedLaunchSettingsProvider"/>
+        /// instead as that provides a joinable variant.
+        /// </remarks>
         IReceivableSourceBlock<ILaunchSettings> SourceBlock { get; }
```

Comment:
Would it make sense to mark this as `Obsolete`? Can you use the new SourceBlock as a replacement for this one?

Reply:
In general, the `SourceBlock` member on the new interface could serve as a replacement for this, yes.

The existing member is a part of a public API (`ILaunchSettingsProvider`) and is definitely consumed outside of our repo. Before we mark it as obsolete we need a public interface to replace it (`IVersionedLaunchSettingsProvider` is internal right now) and guidance about how to consume it properly (if you're piping the settings into another data flow block, you need to join the flows properly, etc.).

Tagging this method as obsolete puts a burden on the consumer, and I don't want to do that until we have those other pieces in place.